Sources/Clappr_iOS/Classes/Plugin/Base/DoubleTapPlugin.swift

```diff
@@ -0,0 +1,70 @@
+import UIKit
+
+public class DoubleTapPlugin: UICorePlugin {
+    
+    var doubleTapGesture: UITapGestureRecognizer!
+    
+    override open var pluginName: String {
+        return String(describing: DoubleTapPlugin.self)
+    }
+    
+    private var activePlayback: Playback? {
+        return core?.activePlayback
+    }
+    
+    private var animatonHandler: DoubleTapAnimation?
+    
+    required init(context: UIObject) {
+        super.init(context: context)
+        animatonHandler = DoubleTapAnimation(core)
+        bindEvents()
+    }
+    
+    required init() {
+        super.init()
+    }
+  
+    private func bindEvents() {
+        stopListening()
+        bindCoreEvents()
+    }
+    
+    private func bindCoreEvents() {
+        guard let core = core else { return }
+        listenTo(core, eventName: Event.didChangeActiveContainer.rawValue) { [weak self] _ in self?.bindEvents() }
+        listenTo(core, eventName: Event.didShowModal.rawValue) { [weak self] _ in self?.removeDoubleTapGesture() }
+        listenTo(core, eventName: Event.didHideModal.rawValue) { [weak self] _ in self?.addDoubleTapGesture() }
+    }
+    
+    func removeDoubleTapGesture() {
+        Logger.logInfo(#function)
```

Comment:
Is the plugin usable without adding and removing the gesture recogniser?
What I want to clarify with this question is if we should let the log, or if we should throw an error telling this should be implemented by descendant classes.

Reply:
That's an excellent question! I wasn't sure about which approach to take.
Throwing an exception seems more reasonable since we want it to crash if its not properly setup. What do you think?